packages/repository/src/relations/has-one/has-one.repository.ts

```diff
@@ -87,4 +112,47 @@ export class DefaultHasOneRepository<
     }
     return found[0];
   }
+
+  async delete(where?: Where<TargetEntity>, options?: Options): Promise<Count> {
+    const targetRepository = await this.getTargetRepository();
+    return targetRepository.deleteAll(
+      constrainWhere(where, this.constraint as Where<TargetEntity>),
```

Comment:
Why is the explicit cast to `Where<TargetEntity>` needed here? Can we find a way that would not require it?

Reply:
> Why is the explicit cast to `Where<TargetEntity>` needed here? Can we find a way that would not require it?

☝️